src/anls_star.py

```diff
@@ -277,7 +311,21 @@ def _levenshtein_distance(s1: str, s2: str):
         return distances[-1]
 
 
-def anls_score(gt, pred):
+@overload
+def anls_score(gt: Any, pred: Any, *, return_gt: Literal[False]) -> float: ...
+
+
+@overload
+def anls_score(
+    gt: Any, pred: Any, *, return_gt: Literal[True]
+) -> tuple[float, Any]: ...
+
+
+@overload
+def anls_score(gt: Any, pred: Any, *, return_gt: bool = False) -> float: ...
+
+
+def anls_score(gt, pred, *, return_gt=False) -> Union[float, tuple[float, Any]]:
```

Comment:
I think(?) usually with overloads you don't specify type-hints on the actual implementation
From the doc (https://docs.python.org/3/library/typing.html#overload) 
![image](https://github.com/deepopinion/anls_star_metric/assets/237550/12e54497-0064-42af-a4b5-e3c8e5d482ea)